cmake/generic.cmake

```diff
@@ -88,7 +88,7 @@
 #
 
 # including binary directory for generated headers.
-include_directories(${CMAKE_BINARY_DIR})
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
Suggest to use `PROJ_BINARY_ROOT` instead, so that we can include the generated `.pb.h` as `#include "paddle/framework/example.pb.h"`.